ocaml/xapi/dbsync.ml

```diff
@@ -59,6 +69,8 @@ let update_env () =
          | _ ->
            error "Multiple pool objects detected -- this should never happen";
            [] in
+       cache_localhost_ref ~__context;
+       set_me_as_resident_on ~__context;
```

Comment:
This wouldn't be needed anymore if you move the place where the localhost ref is set, like @robhoes said above.

Reply:
my bad, yes